diff --git a/bazel/dependency_imports.bzl b/bazel/dependency_imports.bzl
index 1bf6c54a25..4e55a129e4 100644
--- a/bazel/dependency_imports.bzl
+++ b/bazel/dependency_imports.bzl
@@ -20,7 +20,7 @@ load("@rules_rust//rust:defs.bzl", "rust_common")
 load("@rules_rust//rust:repositories.bzl", "rules_rust_dependencies", "rust_register_toolchains", "rust_repository_set")
 
 # go version for rules_go
-GO_VERSION = "1.23.1"
+GO_VERSION = "host"
 
 JQ_VERSION = "1.7"
 YQ_VERSION = "4.24.4"
