diff --git a/bazel/repositories_extra.bzl b/bazel/repositories_extra.bzl
--- a/bazel/repositories_extra.bzl
+++ b/bazel/repositories_extra.bzl
@@ -38,5 +38,6 @@ def envoy_dependencies_extra(
     python_register_toolchains(
         name = "python%s" % _python_minor_version(python_version),
         python_version = python_version,
         ignore_root_user_error = ignore_root_user_error,
+        register_toolchains = False,
     )
