diff --git MODULE.bazel MODULE.bazel
index cc60d224..e83fa2f6 100644
--- MODULE.bazel
+++ MODULE.bazel
@@ -42 +42,11 @@ bazel_dep(
 )
+
+bazel_dep(
+    name = "gloop",
+    repo_name = "do_not_use_for_gloop_visibility_only",
+)
+git_override(
+    module_name = "gloop",
+    remote = "https://github.com/abseil/gloop",
+    commit = "8eead4aaf1eb3685921b09e4fc03a69a31e35d53",
+)
diff --git absl/base/BUILD.bazel absl/base/BUILD.bazel
index 53bac42e..f70c0ca6 100644
--- absl/base/BUILD.bazel
+++ absl/base/BUILD.bazel
@@ -59,2 +59,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -74,2 +75,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -91,2 +93,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -152,2 +155,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -177,2 +181,3 @@ cc_library(
         "//absl/base:__pkg__",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -207,2 +212,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -289,2 +295,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -469,2 +476,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -483,2 +491,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -565,2 +574,3 @@ cc_library(
         "//absl/base:__pkg__",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -600,2 +610,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -871,2 +882,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -970,2 +982,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -1031,2 +1044,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -1061,2 +1075,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
diff --git absl/container/BUILD.bazel absl/container/BUILD.bazel
index a386a537..bd6df665 100644
--- absl/container/BUILD.bazel
+++ absl/container/BUILD.bazel
@@ -233,2 +233,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -450,2 +451,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -951,2 +953,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
@@ -969,2 +972,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
@@ -986,2 +990,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
@@ -1003,2 +1008,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
@@ -1022,2 +1028,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
@@ -1038,2 +1045,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
@@ -1054,2 +1062,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
@@ -1071,2 +1080,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
@@ -1250,2 +1260,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
diff --git absl/debugging/BUILD.bazel absl/debugging/BUILD.bazel
index e6f11e84..5b7480b2 100644
--- absl/debugging/BUILD.bazel
+++ absl/debugging/BUILD.bazel
@@ -160,2 +160,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/log/internal:__pkg__",
@@ -223,3 +224,6 @@ cc_library(
     linkopts = ABSL_DEFAULT_LINKOPTS,
-    visibility = ["//absl:friends"],
+    visibility = [
+        "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
+    ],
     deps = [
@@ -241,2 +245,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/container:__pkg__",
diff --git absl/flags/BUILD.bazel absl/flags/BUILD.bazel
index b5ecb7f7..c7163a6d 100644
--- absl/flags/BUILD.bazel
+++ absl/flags/BUILD.bazel
@@ -46,2 +46,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/flags:__pkg__",
@@ -66,2 +67,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/flags:__pkg__",
@@ -134,2 +136,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -173,2 +176,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/flags:__pkg__",
@@ -274,2 +278,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl/flags:__pkg__",
diff --git absl/log/internal/BUILD.bazel absl/log/internal/BUILD.bazel
index bc473308..53bce896 100644
--- absl/log/internal/BUILD.bazel
+++ absl/log/internal/BUILD.bazel
@@ -29,2 +29,3 @@ package(
         ":internal_users",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -112,2 +113,3 @@ cc_library(
         "//absl/log:__pkg__",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -125,2 +127,3 @@ cc_library(
         "//absl/status:__pkg__",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -160,2 +163,3 @@ cc_library(
         "//absl/log:__pkg__",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -193,2 +197,3 @@ cc_library(
         "//absl/log:__pkg__",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -312,2 +317,3 @@ cc_library(
         ":structured_proto_users",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -386,2 +392,3 @@ cc_library(
         ":structured_proto_users",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -407,2 +414,3 @@ cc_library(
         ":structured_proto_users",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -472,2 +480,3 @@ cc_library(
         "//absl/status:__pkg__",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -569,2 +578,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
diff --git absl/meta/BUILD.bazel absl/meta/BUILD.bazel
index 6e52c6ec..9aef4b2a 100644
--- absl/meta/BUILD.bazel
+++ absl/meta/BUILD.bazel
@@ -45,2 +45,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -71,2 +72,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
diff --git absl/profiling/BUILD.bazel absl/profiling/BUILD.bazel
index 34e3d254..b058c8f3 100644
--- absl/profiling/BUILD.bazel
+++ absl/profiling/BUILD.bazel
@@ -43,2 +43,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -81,2 +82,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -110,2 +112,3 @@ cc_library(
     visibility = [
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
         "//absl:__subpackages__",
diff --git absl/random/internal/BUILD.bazel absl/random/internal/BUILD.bazel
index 56377ed1..c229570a 100644
--- absl/random/internal/BUILD.bazel
+++ absl/random/internal/BUILD.bazel
@@ -32,2 +32,3 @@ default_package_visibility = [
     "//absl:friends",
+    "@do_not_use_for_gloop_visibility_only//:__subpackages__",
 ]
diff --git absl/strings/BUILD.bazel absl/strings/BUILD.bazel
index def78d53..70449873 100644
--- absl/strings/BUILD.bazel
+++ absl/strings/BUILD.bazel
@@ -437,2 +437,3 @@ cc_library(
         "//absl/status:__pkg__",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -1601,2 +1602,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
diff --git absl/synchronization/BUILD.bazel absl/synchronization/BUILD.bazel
index 583eff2d..ffdf46e1 100644
--- absl/synchronization/BUILD.bazel
+++ absl/synchronization/BUILD.bazel
@@ -68,2 +68,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -299,2 +300,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
@@ -348,2 +350,3 @@ cc_library(
         "//absl:friends",
+        "@do_not_use_for_gloop_visibility_only//:__subpackages__",
     ],
