diff --git a/ossm/vendor/proxy_wasm_cpp_host/bazel/BUILD b/ossm/vendor/proxy_wasm_cpp_host/bazel/BUILD
index 15f323fa7..23f677cb3 100644
--- a/ossm/vendor/proxy_wasm_cpp_host/bazel/BUILD
+++ b/ossm/vendor/proxy_wasm_cpp_host/bazel/BUILD
@@ -52,21 +52,8 @@ config_setting(
 )

 config_setting(
-    name = "requested_crypto_system",
-    values = {"define": "crypto=system"},
-)
-
-config_setting(
-    name = "linux_s390x",
-    values = {"cpu": "s390x"},
-)
-
-selects.config_setting_group(
     name = "crypto_system",
-    match_any = [
-        ":requested_crypto_system",
-        ":linux_s390x",
-    ],
+    values = {"define": "crypto=system"},
 )

 exports_files(["tsan_suppressions.txt"])
