diff --git a/ossm/vendor/proxy_wasm_cpp_host/bazel/BUILD b/ossm/vendor/proxy_wasm_cpp_host/bazel/BUILD
index 58b4aed789..4db441139e 100644
--- a/ossm/vendor/proxy_wasm_cpp_host/bazel/BUILD
+++ b/ossm/vendor/proxy_wasm_cpp_host/bazel/BUILD
@@ -53,21 +53,8 @@ config_setting(
 )
 
 config_setting(
-    name = "requested_crypto_system",
-    values = {"define": "crypto=system"},
-)
-
-config_setting(
-    name = "linux_s390x",
-    values = {"cpu": "s390x"},
-)
-
-selects.config_setting_group(
     name = "crypto_system",
-    match_any = [
-        ":requested_crypto_system",
-        ":linux_s390x",
-    ],
+    values = {"define": "crypto=system"},
 )
 
 # V8 library with pointer compression disabled using V8's Starlark transition.
diff --git a/ossm/vendor/proxy_wasm_cpp_host/src/pairs_util.cc b/ossm/vendor/proxy_wasm_cpp_host/src/pairs_util.cc
index b8121882fd..5b65162562 100644
--- a/ossm/vendor/proxy_wasm_cpp_host/src/pairs_util.cc
+++ b/ossm/vendor/proxy_wasm_cpp_host/src/pairs_util.cc
@@ -65,7 +65,7 @@ bool PairsUtil::marshalPairs(const Pairs &pairs, char *buffer, size_t size) {
 
   // Write number of pairs.
   uint32_t num_pairs =
-      htowasm(pairs.size(), contextOrEffectiveContext() != nullptr
+      htowasm(static_cast<uint32_t>(pairs.size()), contextOrEffectiveContext() != nullptr
                                 ? contextOrEffectiveContext()->wasmVm()->usesWasmByteOrder()
                                 : false);
   if (pos + sizeof(uint32_t) > end) {
@@ -77,7 +77,7 @@ bool PairsUtil::marshalPairs(const Pairs &pairs, char *buffer, size_t size) {
   for (const auto &p : pairs) {
     // Write name length.
     uint32_t name_len =
-        htowasm(p.first.size(), contextOrEffectiveContext() != nullptr
+        htowasm(static_cast<uint32_t>(p.first.size()), contextOrEffectiveContext() != nullptr
                                     ? contextOrEffectiveContext()->wasmVm()->usesWasmByteOrder()
                                     : false);
     if (pos + sizeof(uint32_t) > end) {
@@ -88,7 +88,7 @@ bool PairsUtil::marshalPairs(const Pairs &pairs, char *buffer, size_t size) {
 
     // Write value length.
     uint32_t value_len =
-        htowasm(p.second.size(), contextOrEffectiveContext() != nullptr
+        htowasm(static_cast<uint32_t>(p.second.size()), contextOrEffectiveContext() != nullptr
                                      ? contextOrEffectiveContext()->wasmVm()->usesWasmByteOrder()
                                      : false);
     if (pos + sizeof(uint32_t) > end) {

