diff --git a/ossm/vendor/quiche/quiche/common/quiche_data_reader.cc b/ossm/vendor/quiche/quiche/common/quiche_data_reader.cc
index 9d900ad88..306384fd7 100644
--- a/ossm/vendor/quiche/quiche/common/quiche_data_reader.cc
+++ b/ossm/vendor/quiche/quiche/common/quiche_data_reader.cc
@@ -66,6 +66,9 @@ bool QuicheDataReader::ReadUInt32(uint32_t* result) {
   }
   if (endianness_ == quiche::NETWORK_BYTE_ORDER) {
     *result = quiche::QuicheEndian::NetToHost32(*result);
+  } else if (endianness_ == quiche::HOST_BYTE_ORDER
+             && quiche::QuicheEndian::HostEndianness == quiche::BIG) {
+    *result = quiche::QuicheEndian::ByteSwap32(*result);
   }
   return true;
 }
@@ -76,6 +79,9 @@ bool QuicheDataReader::ReadUInt64(uint64_t* result) {
   }
   if (endianness_ == quiche::NETWORK_BYTE_ORDER) {
     *result = quiche::QuicheEndian::NetToHost64(*result);
+  } else if (endianness_ == quiche::HOST_BYTE_ORDER
+             && quiche::QuicheEndian::HostEndianness == quiche::BIG) {
+    *result = quiche::QuicheEndian::ByteSwap64(*result);
   }
   return true;
 }
@@ -86,7 +92,13 @@ bool QuicheDataReader::ReadBytesToUInt64(size_t num_bytes, uint64_t* result) {
     return false;
   }
   if (endianness_ == quiche::HOST_BYTE_ORDER) {
-    return ReadBytes(result, num_bytes);
+    if (!ReadBytes(result, num_bytes)) {
+      return false;
+    }
+    if (quiche::QuicheEndian::HostEndianness == quiche::BIG) {
+      *result = quiche::QuicheEndian::ByteSwap64(*result);
+    }
+    return true;
   }

   if (!ReadBytes(reinterpret_cast<char*>(result) + sizeof(*result) - num_bytes,
diff --git a/ossm/vendor/quiche/quiche/common/quiche_data_writer.cc b/ossm/vendor/quiche/quiche/common/quiche_data_writer.cc
index c439e72ba..cd8e6ca33 100644
--- a/ossm/vendor/quiche/quiche/common/quiche_data_writer.cc
+++ b/ossm/vendor/quiche/quiche/common/quiche_data_writer.cc
@@ -35,6 +35,9 @@ bool QuicheDataWriter::WriteUInt8(uint8_t value) {
 bool QuicheDataWriter::WriteUInt16(uint16_t value) {
   if (endianness_ == quiche::NETWORK_BYTE_ORDER) {
     value = quiche::QuicheEndian::HostToNet16(value);
+  } else if (endianness_ == quiche::HOST_BYTE_ORDER
+             && quiche::QuicheEndian::HostEndianness == quiche::BIG) {
+    value = quiche::QuicheEndian::ByteSwap16(value);
   }
   return WriteBytes(&value, sizeof(value));
 }
@@ -42,6 +45,9 @@ bool QuicheDataWriter::WriteUInt16(uint16_t value) {
 bool QuicheDataWriter::WriteUInt32(uint32_t value) {
   if (endianness_ == quiche::NETWORK_BYTE_ORDER) {
     value = quiche::QuicheEndian::HostToNet32(value);
+  } else if (endianness_ == quiche::HOST_BYTE_ORDER
+             && quiche::QuicheEndian::HostEndianness == quiche::BIG) {
+    value = quiche::QuicheEndian::ByteSwap32(value);
   }
   return WriteBytes(&value, sizeof(value));
 }
@@ -49,6 +55,9 @@ bool QuicheDataWriter::WriteUInt32(uint32_t value) {
 bool QuicheDataWriter::WriteUInt64(uint64_t value) {
   if (endianness_ == quiche::NETWORK_BYTE_ORDER) {
     value = quiche::QuicheEndian::HostToNet64(value);
+  } else if (endianness_ == quiche::HOST_BYTE_ORDER
+             && quiche::QuicheEndian::HostEndianness == quiche::BIG) {
+    value = quiche::QuicheEndian::ByteSwap64(value);
   }
   return WriteBytes(&value, sizeof(value));
 }
@@ -58,6 +67,9 @@ bool QuicheDataWriter::WriteBytesToUInt64(size_t num_bytes, uint64_t value) {
     return false;
   }
   if (endianness_ == quiche::HOST_BYTE_ORDER) {
+    if (quiche::QuicheEndian::HostEndianness == quiche::BIG) {
+      value = quiche::QuicheEndian::ByteSwap64(value);
+    }
     return WriteBytes(&value, num_bytes);
   }

diff --git a/ossm/vendor/quiche/quiche/common/quiche_endian.h b/ossm/vendor/quiche/quiche/common/quiche_endian.h
index ee5999fb2..c90f12951 100644
--- a/ossm/vendor/quiche/quiche/common/quiche_endian.h
+++ b/ossm/vendor/quiche/quiche/common/quiche_endian.h
@@ -19,28 +19,65 @@ enum Endianness {
   HOST_BYTE_ORDER      // little endian
 };

+enum HostEndianness {
+  BIG,
+  LITTLE
+};
+
 // Provide utility functions that convert from/to network order (big endian)
 // to/from host order (little endian).
 class QUICHE_EXPORT QuicheEndian {
  public:
-  // Convert |x| from host order (little endian) to network order (big endian).
-#if defined(__clang__) || \
-    (defined(__GNUC__) && \
-     ((__GNUC__ == 4 && __GNUC_MINOR__ >= 8) || __GNUC__ >= 5))
-  static uint16_t HostToNet16(uint16_t x) { return __builtin_bswap16(x); }
-  static uint32_t HostToNet32(uint32_t x) { return __builtin_bswap32(x); }
-  static uint64_t HostToNet64(uint64_t x) { return __builtin_bswap64(x); }
+  // Get host machine endianness
+#if defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && \
+    __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+  static const quiche::HostEndianness HostEndianness = quiche::BIG;
+#else
+  static const quiche::HostEndianness HostEndianness = quiche::LITTLE;
+#endif
+
+  // Convert byte order of |x|.
+ #if defined(__clang__) || \
+  (defined(__GNUC__) && \
+   ((__GNUC__ == 4 && __GNUC_MINOR__ >= 8) || __GNUC__ >= 5))
+  static uint16_t ByteSwap16(uint16_t x) { return __builtin_bswap16(x); }
+  static uint32_t ByteSwap32(uint32_t x) { return __builtin_bswap32(x); }
+  static uint64_t ByteSwap64(uint64_t x) { return __builtin_bswap64(x); }
 #else
-  static uint16_t HostToNet16(uint16_t x) { return PortableByteSwap(x); }
-  static uint32_t HostToNet32(uint32_t x) { return PortableByteSwap(x); }
-  static uint64_t HostToNet64(uint64_t x) { return PortableByteSwap(x); }
+  static uint16_t ByteSwap16(uint16_t x) { return PortableByteSwap(x); }
+  static uint32_t ByteSwap32(uint32_t x) { return PortableByteSwap(x); }
+  static uint64_t ByteSwap64(uint64_t x) { return PortableByteSwap(x); }
 #endif

-  // Convert |x| from network order (big endian) to host order (little endian).
+  // Convert |x| from host order to network order (big endian).
+#if defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && \
+    __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+  static uint16_t HostToNet16(uint16_t x) { return x; }
+  static uint32_t HostToNet32(uint32_t x) { return x; }
+  static uint64_t HostToNet64(uint64_t x) { return x; }
+#else
+  static uint16_t HostToNet16(uint16_t x) { return ByteSwap16(x); }
+  static uint32_t HostToNet32(uint32_t x) { return ByteSwap32(x); }
+  static uint64_t HostToNet64(uint64_t x) { return ByteSwap64(x); }
+#endif
+
+  // Convert |x| from network order (big endian) to host order.
   static uint16_t NetToHost16(uint16_t x) { return HostToNet16(x); }
   static uint32_t NetToHost32(uint32_t x) { return HostToNet32(x); }
   static uint64_t NetToHost64(uint64_t x) { return HostToNet64(x); }

+  // Convert |x| from host order to little endian order.
+#if defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && \
+    __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+  static uint16_t HostToLittleEndian16(uint16_t x) { return ByteSwap16(x); }
+  static uint32_t HostToLittleEndian32(uint32_t x) { return ByteSwap32(x); }
+  static uint64_t HostToLittleEndian64(uint64_t x) { return ByteSwap64(x); }
+#else
+  static uint16_t HostToLittleEndian16(uint16_t x) { return x; }
+  static uint32_t HostToLittleEndian32(uint32_t x) { return x; }
+  static uint64_t HostToLittleEndian64(uint64_t x) { return x; }
+#endif
+
   // Left public for tests.
   template <typename T>
   static T PortableByteSwap(T input) {
